OneCommunityGlobal/HGNRest
# Anoushka application search

# Description
![image](https://github.com/user-attachments/assets/096787fa-5638-4aca-87f6-74d4577cf6f5)
- Search API was already implemented in existing code so I worked on 2,3 and 4.

## Related PRS (if any):
This is a backend PR. You can use Postman to test functionality.
…

## Main changes explained:
- Added a summaries API which fetch job summaries with pagination, search, filtering, and sorting when provided a job title or category. The summary API only shows title,category,location,description,datePosted,featured columns
- Added a job suggestions API which gives suggestions for a drop down menu given s job title search query.
- Added a reset Filter API which resets filters and returns all jobs sorted by title, date posted and featured or not

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Go to Postman and test out the endpoints
4.  Use GET `http://localhost:4500/api/jobs/summaries?search=Analyst` for testing Summary API
5. Use GET `http://localhost:4500/api/jobs/suggestions?query=an` for testing the Drop down suggestion API
6. Use GET `http://localhost:4500/api/jobs/reset-filters` for reset filter 


## Screenshots or videos of changes:

https://github.com/user-attachments/assets/2e88366f-4ee8-4290-ae4a-cb2df694630a
